packages/fxa-settings/src/pages/Signup/container.spec.tsx

```diff
@@ -0,0 +1,439 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This is our first POC for container tests. There's more comments than normal in this file, because it is
+ * establishing the patterns going forward for future container tests. Comment blocks starting with TIP give
+ * some guidance around common testing practices
+ *
+ * A couple things to keep in mind:
+ *  - These tests should only be validating logic inside the container!
+ *  - These tests will need to mock the containers dependencies
+ *  - These tests should not:
+ *    - Validate logic in the presentation layer
+ *    - Validate logic used in hooks
+ *    - Validate functionality of third party libraries
+ *    - Require any external APIs or database
+ *
+ * The basic steps should e as follows:
+ *  1. Identify what needs to be mocked and import modules
+ *  2. Set up functions that initialize mocks and can be reused or overridden, preferably using jest.spyOn.
+ *  4. Write tests, be sure to mocks & spies are reset prior to each suite.
+ *  5. Assert on mocked functions and spies
+ */
+
+import React from 'react';
+
+// TIP - Import modules for mocking. Not that `* as` lend themselves to using jest.spyOn.
+import * as SignupModule from './index';
+import * as ModelsModule from '../../models';
+import * as LoadingSpinnerModule from 'fxa-react/components/LoadingSpinner';
+import * as ApolloModule from '@apollo/client';
+import * as UseValidateModule from '../../lib/hooks/useValidate';
+import * as FirefoxModule from '../../lib/channels/firefox';
+import * as CryptoModule from 'fxa-auth-client/lib/crypto';
+import * as ReachRouterModule from '@reach/router';
+import * as ReactUtils from 'fxa-react/lib/utils';
+
+// Typical imports
+import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
+import { screen } from '@testing-library/react';
+import SingupContainer, { SignupContainerIntegration } from './container';
+import { IntegrationType, useAuthClient } from '../../models';
+import { MozServices } from '../../lib/types';
+import { FirefoxCommand } from '../../lib/channels/firefox';
```

Comment:
What dis? Aren't we already doing this above w/ L34?
And I guess same comment applies above to L43-vs-L30.

Reply:
It's the direct import. We could referenernce the module instead on line 34, or just do this. IMO either works, but if it looks off to you, I'd be happy to change it.